ksiren/src/main/java/com/brightspace/ksiren/KSirenRequestBuilder.kt

```diff
@@ -15,21 +15,14 @@ package com.brightspace.ksiren
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-abstract class KSirenRequestBuilder<T>(val action: Action) {
-	protected val fieldValues: MutableMap<String, String> = mutableMapOf()
+abstract class KSirenRequestBuilder<T>(action: ActionBase) {
+	private val mutableAction = MutableAction(action)
 
-	init {
-		action.fields.forEach {
-			field ->
-
-			field.value?.let {
-				fieldValues[field.name] = it
-			}
-		}
-	}
+	val action: ActionBase
+		get() = mutableAction
 
 	fun addFieldValue(name: String, value: String): KSirenRequestBuilder<T> {
-		fieldValues.put(name, value)
+		mutableAction.fields.find { it.name == name }?.value = value
```

Comment:
These do different things, don't they?  fieldValues.put(x, y) will add the key if it doesn't exist.  If your name doesn't exist, nothing is added.

Reply:
In Siren, action fields are controls/affordances that follow HTML5 input semantics, so a client creating a new field arbitrarily wouldn't make much sense, and also isn't what's being done by consumers using this library today, so we don't have to account for deviations.

It could actually be more strict, eg. to not allow writing to fields of type `hidden` (`hasField` should do this also), but I'd like to open as few cans of worms as possible.